app/src/main/java/com/it/hungvt/movieapp/screen/movie/MovieAdapter.java

```diff
@@ -0,0 +1,130 @@
+package com.it.hungvt.movieapp.screen.movie;
+
+import android.content.Context;
+import android.databinding.DataBindingUtil;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+import android.widget.Filter;
+import android.widget.Filterable;
+import com.it.hungvt.movieapp.BaseRecyclerViewAdapter;
+import com.it.hungvt.movieapp.R;
+import com.it.hungvt.movieapp.data.model.Movie;
+import com.it.hungvt.movieapp.databinding.ItemMovieBinding;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Binding data on Movie screen
+ */
+
+public class MovieAdapter extends BaseRecyclerViewAdapter<MovieAdapter.ItemViewHolder>
+        implements Filterable {
+
+    private List<Movie> mMovies;
+    private ItemMovieBinding mBinding;
+    private BaseRecyclerViewAdapter.OnRecyclerViewItemClickListener<Movie> mItemClickListener;
+    private List<Movie> mFilterListMovie;
+
+    public MovieAdapter(@NonNull Context context, List<Movie> movies) {
+        super(context);
+        mMovies = movies;
+        mFilterListMovie = movies;
+    }
+
+    @Override
+    public ItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        LayoutInflater mInflater = LayoutInflater.from(parent.getContext());
+
+        mBinding = DataBindingUtil.inflate(mInflater, R.layout.item_movie, parent, false);
+
+        return new ItemViewHolder(mBinding, mItemClickListener);
+    }
+
+    @Override
+    public void onBindViewHolder(ItemViewHolder holder, int position) {
+
+        holder.bind(mMovies.get(position));
+    }
+
+    @Override
+    public int getItemCount() {
+        return (mMovies == null) ? 0 : mMovies.size();
+    }
+
+    public void setItemClickListener(OnRecyclerViewItemClickListener<Movie> itemClickListener) {
+        mItemClickListener = itemClickListener;
+    }
+
+    public List<Movie> getMovies() {
+        return mMovies;
+    }
+
+    public void updateMovies(List<Movie> movies) {
+        if (mMovies == null) {
+            return;
+        }
+
+        mMovies.addAll(movies);
+        notifyDataSetChanged();
+    }
+
+    @Override
+    public Filter getFilter() {
+        return new Filter() {
+            @Override
+            protected FilterResults performFiltering(CharSequence constraint) {
+
+                String charString = constraint.toString();
+                if (charString.isEmpty()) {
+
+                    mFilterListMovie = mMovies;
+                } else {
+
+                    List<Movie> filteredList = new ArrayList<>();
+                    for (Movie movie : mMovies) {
+                        if (movie.getTitle().toLowerCase().contains(charString)
+                                || (movie.getVoteAverage() + "").toLowerCase()
```

Comment:
WARNING: [DefaultLocale] Implicitly using the default locale is a common source of bugs: Use `toLowerCase(Locale)` instead. For strings meant to be internal use `Locale.ROOT`, otherwise `Locale.getDefault()`.
 Calling `String#toLowerCase()` or `#toUpperCase()` **without specifying an explicit locale** is a common source of bugs. The reason for that is that those methods will use the current locale on the user's device, and even though the code appears to work correctly when you are developing the app, it will fail in some locales. For example, in the Turkish locale, the uppercase replacement for `i` is **not** `I`.

If you want the methods to just perform ASCII replacement, for example to convert an enum name, call `String#toUpperCase(Locale.US)` instead. If you really want to use the current locale, call `String#toUpperCase(Locale.getDefault())` instead.